spi/src/main/java/org/opensearch/security/spi/resources/ResourceSharingExtension.java

```diff
@@ -16,20 +20,17 @@
 public interface ResourceSharingExtension {
 
     /**
-     * Type of the resource
-     * @return a string containing the type of the resource. A qualified class name can be supplied here.
+     * Returns the set of {@link ResourceProvider} instances for the resources defined by the plugin.
+     * Only in the case where plugin defines multiple resources, will there be more than one resource provider
+    *
+     * @return the set of ResourceProvider instances
      */
-    String getResourceType();
+    Set<ResourceProvider> getResourceProviders();
 
     /**
-     * The index where resource is stored
-     * @return the name of the parent index where resource is stored
+     * Assigns the ResourceSharingClient to the resource plugin. Plugins can then utilize this to call the methods for access control.
+     * @param client the ResourceSharingClient instance
      */
-    String getResourceIndex();
+    void assignResourceSharingClient(ResourceSharingClient client);
```

Comment:
For plugins that implement this extension, can't we rely on the the ResourceSharingClient never being provided as having either 1) security uninstalled, 2) security disabled or 3) this feature disabled?

I see that there are checks added in the AD PR to see if security is disabled, but I don't think the checks should be present in those repos.

Reply:
> I see that there are checks added in the AD PR to see if security is disabled, but I don't think the checks should be present in those repos.

The checks are meant to be for the feature to be disabled. If you do see a security enabled/disabled check then it is a miss on my end. Would you mind pointing me to the exact code change?